src/mixins/nr-affiliation-mixin.ts

```diff
@@ -115,14 +115,27 @@ export class NrAffiliationMixin extends Mixins(CommonMixin) {
     const legalType = this.isBenefitCompany(nr) ? 'BEN' : nr.legalType
     const nrNumber = nr.nrNum
 
-    return {
+    const businessRequest = {
       filing: {
         business: { legalType },
-        header: { accountId, name },
-        incorporationApplication: !this.isFirm(nr) ? { nameRequest: { legalType, nrNumber } } : undefined,
-        registration: this.isFirm(nr) ? { nameRequest: { legalType, nrNumber } } : undefined
+        header: { accountId, name }
       }
     } as BusinessRequest
+
+    if (!this.isFirm(nr)) {
+      businessRequest.filing.incorporationApplication = {
+        nameRequest: { legalType, nrNumber }
+      }
+    }
+
+    if (this.isFirm(nr)) {
+      businessRequest.filing.registration = {
+        business: { natureOfBusiness: nr.natureOfBusiness },
+        nameRequest: { legalType, nrNumber }
+      }
```

Comment:
This is now consistent with the [initial draft that auth-web creates](https://github.com/bcgov/sbc-auth/blob/f695e4ea7a025c4f929b3503a5e54fc71fbd574f/auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue#L402).

Reply:
PS This is the root cause of the subject bug ticket, but I also updated Create UI to handle the case of no business object (since there are suitable defaults for its fields anyway).

Reply:
Looks good, thanks bud